app/controllers/v3/routes_controller.rb

```diff
@@ -130,6 +131,27 @@ def share_routes
     unprocessable!(e.message)
   end
 
+  def unshare_routes
+    FeatureFlag.raise_unless_enabled!(:route_sharing)
+    unauthorized! unless permission_queryer.can_manage_apps_in_space?(route.space.guid)
+
+    space_guid = hashed_params[:space_guid]
+
+    target_space = Space.first(guid: space_guid)
+    resource_not_found!(:space) unless target_space && permission_queryer.can_read_from_space?(space_guid, target_space.organization.guid)
+
+    if permission_queryer.can_manage_apps_in_space?(target_space.guid) == false
+      unprocessable!("Unable to unshare route '#{route.uri}' from space '#{target_space.name}'. Ensure that the space and its containing org are not suspended.")
+    end
+
+    unshare = RouteUnshare.new
+    unshare.delete(route, target_space, user_audit_info)
+
```

Comment:
@MerricdeLauney So, this reads a little weirdly: "`unshare.delete`", but I couldn't come up with a better set of names after a few minutes of thinking. Alternatives are very welcome.